src/bin/check.rs

```diff
@@ -106,6 +111,17 @@ pub fn execute(options: Options, config: &mut Config) -> CliResult {
                                     &options.flag_exclude,
                                     &options.flag_package)?;
 
+    let test = options.flag_tests ||
```

Comment:
This logic may cause subtle oddities, although that's maybe ok for now? but for example:

```
cargo check --tests --bin foo
```

I think that'll check the binary `foo` in "test mode" rather than "do you compile mode", right?

Reply:
You're right.  The more I think about it, it is probably wrong.  `check --tests` should not check bins with the test profile.  I'll fix it.